qutip/solver/correlation.py

```diff
@@ -0,0 +1,497 @@
+        System Hamiltonian.
+    state0 : Qobj
+        Initial state density matrix :math:`\rho(t_0)` or state vector
+        :math:`\psi(t_0)`. If 'state0' is 'None', then the steady state will
+        be used as the initial state. The 'steady-state' is only implemented
+        for the ``me`` and ``es`` solvers if ``c_ops`` are provided.
+    taulist : array_like
+        List of times for :math:`\tau`. taulist must be positive and contain
+        the element `0`.
+    c_ops : list of {Qobj, QobjEvo}
+        List of collapse operators
+    a_op : Qobj, QobjEvo
+        Operator A.
+    b_op : Qobj, QobjEvo
+        Operator B.
+    c_op : Qobj, QobjEvo
+        Operator C.
+    solver : str
+        choice of solver (`me` for master-equation, and `es` for exponential
+        series).
+    options : dict
+        Options for the solver.
+
+    Returns
+    -------
+    corr_vec : array
+        An array of correlation values for the times specified by `taulist`
+
+    References
+    ----------
+    See, Gardiner, Quantum Noise, Section 5.2.
+
+    """
+    solver = _make_solver(H, c_ops, args, options, solver)
+    if state0 is None:
+        state0 = steadystate(H, c_ops)
+    return correlation_3op(solver, state0, [0], taulist, a_op, b_op, c_op)[0]
+
+
+def correlation_3op_2t(H, state0, tlist, taulist, c_ops, a_op, b_op, c_op,
+                       solver="me", args={},
+                       options={}):
+    r"""
+    Calculate the three-operator two-time correlation function:
+    :math:`\left<A(t)B(t+\tau)C(t)\right>` along two time axes using the
+    quantum regression theorem and the evolution solver indicated by the
+    `solver` parameter.
+
+    Note: it is not possibly to calculate a physically meaningful correlation
+    of this form where :math:`\tau<0`.
+
+    Parameters
+    ----------
+    H : Qobj, QobjEvo
+        System Hamiltonian.
+    state0 : Qobj
+        Initial state density matrix :math:`\rho(t_0)` or state vector
+        :math:`\psi(t_0)`. If 'state0' is 'None', then the steady state will
+        be used as the initial state. The 'steady-state' is only implemented
+        for the ``me`` and ``es`` solvers if ``c_ops`` are provided.
+    tlist : array_like
+        List of times for :math:`t`. tlist must be positive and contain the
+        element `0`. When taking steady-steady correlations only one tlist
+        value is necessary, i.e. when :math:`t \rightarrow \infty`; here
+        tlist is automatically set, ignoring user input.
+    taulist : array_like
+        List of times for :math:`\tau`. taulist must be positive and contain
+        the element `0`.
+    c_ops : list of {Qobj, QobjEvo}
+        List of collapse operators
+    a_op : Qobj, QobjEvo
+        Operator A.
+    b_op : Qobj, QobjEvo
+        Operator B.
+    c_op : Qobj, QobjEvo
+        Operator C.
+    solver : str
+        Choice of solver (`me` for master-equation, and `es` for exponential
+        series).
```

Comment:
I think we can give a little bit more information about the solver as string:
```suggestion
        Choice of solver (`me` for master-equation, and `es` for exponential
        series). Both solver use `MeSolver` internally. However, `es`
        method sets the integrator to be 'diag'. If a custom integrator is to be
        employed, use 'me' option.
```

Reply:
Reworded in my words